ted/src/main/java/frc/robot/commands/AllignRelativeToTagCommand.java

```diff
@@ -0,0 +1,129 @@
+      // find the difference between current pose in vision space and target pose in vision space
+      Twist2d desiredTwist = robotPoseInVisionSpace.log(targetPoseInVisionSpace);
+      startingPositionInRobotSpace = drivetrainsubsystem.getRobotPosition();
+      // apply the twist to the robot's current position
+      targetPositionInRobotSpace = startingPositionInRobotSpace.exp(desiredTwist);
+    }
+    else {
+      System.out.println("no tag in sight");
+      done = true;
+    }
+
+    drivetrainsubsystem.drive(new ChassisSpeeds(0, 0, 0));
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute(){
+      startingPositionInRobotSpace = drivetrainsubsystem.getRobotPosition();
+      double xVelocity = 0.0;
+      double yVelocity = 0.0;
+      double rotVelocity = 0.0;
+
+      if(Math.abs(targetPositionInRobotSpace.getX() - startingPositionInRobotSpace.getX()) >= transformTolerance){
+        xVelocity = xPID.calculate(startingPositionInRobotSpace.getX(), targetPositionInRobotSpace.getX());
+        xVelocity = -1 * MotorUtils.clamp(xVelocity, -velocityFactor, velocityFactor);
+      }
+
+      if(Math.abs(targetPositionInRobotSpace.getY() - startingPositionInRobotSpace.getY()) >= transformTolerance){
+        yVelocity = yPID.calculate(startingPositionInRobotSpace.getY(), targetPositionInRobotSpace.getY());
+        yVelocity = -1 * MotorUtils.clamp(yVelocity, -velocityFactor, velocityFactor);
+      }
+
+      if(Math.abs(targetPositionInRobotSpace.getRotation().getDegrees() - startingPositionInRobotSpace.getRotation().getDegrees()) >= rotationTolerance){
+        rotVelocity = rotationPID.calculate(startingPositionInRobotSpace.getRotation().getDegrees(), targetPositionInRobotSpace.getRotation().getDegrees());
+        rotVelocity = -1 * MotorUtils.clamp(rotVelocity, -velocityFactor, velocityFactor);
+      }
+
+      System.out.println("xVelocity: " + xVelocity);
+      System.out.println("yVelocity: " + yVelocity);
+      System.out.println("rotVelocity: " + rotVelocity);
+      drivetrainsubsystem.drive(new ChassisSpeeds(xVelocity, yVelocity, rotVelocity));
+    }
+
+
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted){
+    done = true;
+    drivetrainsubsystem.drive(new ChassisSpeeds(0, 0, 0));
+  }
+
+  // Returns true when the command should end.
+  @Override
+  public boolean isFinished(){
+    startingPositionInRobotSpace = drivetrainsubsystem.getRobotPosition();
+    //todo
+    // for the rotation, you need to wrap the angle around using angleModulus. angleModulus operates in radians, 
+    // so you will need to wrap the angle in radians and then convert to degrees.
+    done = Units.radiansToDegrees(Math.abs(MathUtil.angleModulus((targetPositionInRobotSpace.getRotation().getRadians() - startingPositionInRobotSpace.getRotation().getDegrees())))) <= rotationTolerance
```

Comment:
shorter lines ... 120 character boundary is customary

Reply:
removed